src/PointwiseFunctions/AnalyticData/GrMhd/MagneticFieldLoop.hpp

```diff
@@ -0,0 +1,218 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Minkowski.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/MakeArray.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+// IWYU pragma:  no_include <pup.h>
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd {
+namespace AnalyticData {
+
+/*!
+ * \brief Analytic initial data for an advecting magnetic field loop.
+ *
+ * This test, originally proposed in \cite Gardiner2005hy and presented in a
+ * slightly modified form by \cite Beckwith2011iy, a region with a circular
+ * cross section with radius `MagneticLoopRadius` is given a non-zero azimuthal
+ * magnetic field of constant magnitude `MagFieldStrength` with zero magnetic
+ * field outside the loop.  This is embedded in an ideal fluid with the given
+ * `AdiabaticIndex`, `RestMassDensity` and `Pressure` with a uniform
+ * `AdvectionVelocity`.  The magnetic field loop should advect across the grid,
+ * maintaining its shape and strength.  The test is typically run in a periodic
+ * domain.
+ *
+ */
+class MagneticFieldLoop {
+ public:
+  using equation_of_state_type = EquationsOfState::IdealFluid<true>;
+  using background_spacetime_type = gr::Solutions::Minkowski<3>;
+
+  /// The pressure throughout the fluid.
+  struct Pressure {
+    using type = double;
+    static constexpr OptionString help = {
+        "The constant pressure throughout the fluid."};
+    static type lower_bound() { return 0.0; }
+  };
+
+  /// The rest mass density throughout the fluid.
+  struct RestMassDensity {
+    using type = double;
+    static constexpr OptionString help = {
+        "The constant density throughout the fluid."};
+    static type lower_bound() { return 0.0; }
+  };
+
+  /// The adiabatic index for the ideal fluid.
+  struct AdiabaticIndex {
+    using type = double;
+    static constexpr OptionString help = {
+        "The adiabatic index for the ideal fluid."};
+    static type lower_bound() { return 1.0; }
+  };
+
+  /// The fluid velocity.
+  struct AdvectionVelocity {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {"The advection velocity."};
+    static type lower_bound() { return {{-1.0, -1.0, -1.0}}; }
+    static type upper_bound() { return {{1.0, 1.0, 1.0}}; }
```

Comment:
I would add a test to prevent all components of the velocity from being set to `1.0` and a test that checks this